c/tests/test_tables.c

```diff
@@ -105,6 +105,102 @@ insert_edge_metadata(tsk_table_collection_t *tables)
+    CU_ASSERT_TRUE(ret);
+
+    // Adding some meat to the tables
+    ret = tsk_node_table_add_row(&tc1.nodes, TSK_NODE_IS_SAMPLE, 0.0, 0, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_node_table_add_row(&tc1.nodes, TSK_NODE_IS_SAMPLE, 1.0, 0, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_individual_table_add_row(&tc1.individuals, 0, NULL, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_population_table_add_row(&tc1.populations, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_edge_table_add_row(&tc1.edges, 0.0, 1.0, 1, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_site_table_add_row(&tc1.sites, 0.2, "A", 1, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+    ret = tsk_mutation_table_add_row(
+        &tc1.mutations, 0, 0, TSK_NULL, TSK_UNKNOWN_TIME, NULL, 0, NULL, 0);
+    CU_ASSERT_FATAL(ret >= 0);
+
+    // Equality of empty vs non-empty
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    // Copying over tc1 to tc2
+    ret = tsk_table_collection_copy(&tc1, &tc2, 0);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+
+    // Equivalence for different metadata/schema
+    ret = tsk_table_collection_set_metadata(
+        &tc1, example_metadata, example_metadata_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    ret = tsk_table_collection_set_metadata(
+        &tc2, example_metadata, example_metadata_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_set_metadata_schema(
+        &tc1, example_metadata_schema, example_metadata_schema_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_FALSE(ret);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_set_metadata_schema(
+        &tc2, example_metadata_schema, example_metadata_schema_length);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
+    CU_ASSERT_TRUE(ret);
+
+    // Ignore provenance
+    ret = tsk_provenance_table_add_row(&tc1.provenances, "time", 4, "record", 6);
+    CU_ASSERT_EQUAL_FATAL(ret, 0);
+    ret = tsk_table_collection_equivalence(&tc1, &tc2, 0);
+    CU_ASSERT_TRUE(ret);
+    ret = tsk_table_collection_equivalence(
+        &tc1, &tc2, TSK_EQ_CMP_METADATA | TSK_EQ_CMP_PROVENANCE);
```

Comment:
and these should just use the `TSK_EQ_CMP_PROVENANCE` flag?

Reply:
thanks!